api_tabular/utils.py

```diff
@@ -37,47 +106,90 @@ async def get_app_version() -> str:
 
 def build_sql_query_string(request_arg: list, page_size: int = None, offset: int = 0) -> str:
     sql_query = []
+    aggregators = defaultdict(list)
     sorted = False
     for arg in request_arg:
-        argument, value = arg.split("=")
-        if "__" in argument:
-            *column_split, comparator = argument.split("__")
-            normalized_comparator = comparator.lower()
-            # handling headers with "__" and special characters
-            # we're escaping the " because they are the encapsulators of the label
-            column = '"{}"'.format("__".join(column_split).replace('"', '\\"'))
-
-            if normalized_comparator == "sort":
-                if value == "asc":
-                    sql_query.append(f"order={column}.asc,__id.asc")
-                elif value == "desc":
-                    sql_query.append(f"order={column}.desc,__id.asc")
-                sorted = True
-            elif normalized_comparator == "exact":
-                sql_query.append(f"{column}=eq.{value}")
-            elif normalized_comparator == "differs":
-                sql_query.append(f"{column}=neq.{value}")
-            elif normalized_comparator == "contains":
-                sql_query.append(f"{column}=ilike.*{value}*")
-            elif normalized_comparator == "in":
-                sql_query.append(f"{column}=in.({value})")
-            elif normalized_comparator == "less":
-                sql_query.append(f"{column}=lte.{value}")
-            elif normalized_comparator == "greater":
-                sql_query.append(f"{column}=gte.{value}")
-            elif normalized_comparator == "strictly_less":
-                sql_query.append(f"{column}=lt.{value}")
-            elif normalized_comparator == "strictly_greater":
-                sql_query.append(f"{column}=gt.{value}")
+        _split = arg.split("=")
+        # filters are expected to have the syntax `<column_name>__<operator>=<value>`
+        if len(_split) == 2:
+            _filter, _sorted = add_filter(*_split)
+            if _filter:
+                sorted = sorted or _sorted
+                sql_query.append(_filter)
+        # aggregators are expected to have the syntax `<column_name>__<operator>`
+        elif len(_split) == 1:
+            column, operator = add_aggregator(_split[0])
+            if column:
+                aggregators[operator].append(column)
+        else:
+            raise ValueError
```

Comment:
Does a `ValueError` raise a HTTP error 400 with `aiohttp`, or would that raise a HTTP 500?

If that raises a HTTP 500, I think we should replace that with `aiohttp.web.HTTPBadRequest`.

Reply:
It does raise `400` [here](https://github.com/datagouv/api-tabular/blob/f5afe16a47992956fe86ab01ea31e34fcd076e35/api_tabular/app.py#L99) 💥

Reply:
Ah perfect sorry didn't see